nuxeo-core/nuxeo-core/src/main/java/org/nuxeo/ecm/core/trash/AbstractTrashService.java

```diff
@@ -305,34 +306,42 @@ public void untrashDocuments(List<DocumentModel> docs) {
      */
     protected static final Pattern COLLISION_PATTERN = Pattern.compile("(.*)\\.[0-9]{13,}");
 
+    @Override
+    public boolean isMangledName(String docName) {
+        Matcher matcher = TRASHED_PATTERN.matcher(docName);
+        return (matcher.matches());
+    }
+
     @Override
     public String mangleName(DocumentModel doc) {
         return doc.getName() + "._" + System.currentTimeMillis() + "_.trashed";
     }
 
     @Override
     public String unmangleName(DocumentModel doc) {
-        String name = doc.getName();
-        Matcher matcher = TRASHED_PATTERN.matcher(name);
-        if (matcher.matches() && matcher.group(1).length() > 0) {
-            name = matcher.group(1);
-            matcher = COLLISION_PATTERN.matcher(name);
-            if (matcher.matches() && matcher.group(1).length() > 0) {
-                CoreSession session = doc.getCoreSession();
-                if (session != null) {
-                    String orig = matcher.group(1);
-                    String parentPath = session.getDocument(doc.getParentRef()).getPathAsString();
+        return unmangleName(doc.getCoreSession(), doc.getName(), doc.getParentRef());
+    }
+
+    @Override
+    public String unmangleName(CoreSession session, String docName, DocumentRef parentRef) {
+        Matcher matcher = TRASHED_PATTERN.matcher(docName);
+        if (matcher.matches() && !StringUtils.isEmpty(matcher.group(1))) {
+            docName = matcher.group(1);
+            matcher = COLLISION_PATTERN.matcher(docName);
+            if (matcher.matches()) {
+                String orig = matcher.group(1);
+                if (session != null && matcher.matches() && !StringUtils.isEmpty(orig)) {
```

Comment:
there is already a matcher.matches() above, there are too many conditions in this method, try to return early